examples/src/Viewer.tsx

```diff
@@ -676,14 +676,15 @@ class Viewer extends React.Component<InputParams, ViewerState> {
         this.setState({ agentColors });
     };
 
-    public setColorSelectionInfo = (colorChange) => {
+    public setColorSelectionInfo = (appliedColors) => {
         this.setState({
             ...this.state,
+            uiDisplayData: appliedColors,
             selectionStateInfo: {
                 hiddenAgents: this.state.selectionStateInfo.hiddenAgents,
                 highlightedAgents:
                     this.state.selectionStateInfo.highlightedAgents,
-                colorChange: colorChange,
+                appliedColors: appliedColors,
             },
```

Comment:
Interesting that state is now duplicated here. I don't have quite enough context on why this would be necessary, or whether this change would imply a similar duplication in the actual simularium-website code. But, particularly if it does, I'd ask a couple questions:

- If these two values got out of sync with each other, would that represent a meaningful UI state or be some sort of error?
- If it would be an error, could you get by with just one copy? e.g. could everything that's currently being driven by `uiDisplayData` instead be driven by `appliedColors`?
- It looks like both these state fields are related to props on `SimulariumViewer` (`onUiDisplayDataChanged`, `selectionStateInfo`). If the type of one now contains the other, could/should those paths be merged in the viewer?

Reply:
Currently no error will throw, but display will be incorrect if these two are out of sync after more than one color selection.

In the website we will be storing two possibilities for color state (so that we can toggle between default and user selections). The `onUiDisplayDataChanged` callback will set the default value, and the user selections will be stored separately. The UI components and the redux selector for the `selectionStateInfo` prop will grab the appropriate one without duplicating the state.

Here we could apply the suggestion from your second bullet, removing the `uiDisplayData` property from `viewerState` and driving everything with `selectionStateInfo.appliedColors`, because we only ever track the current state and never re-apply the default. Not duplicating state = good. But we will be slightly less in sync with naming/conventions of website.

In regards to merging the paths related to `onUiDisplayDataChanged` and `selectionStateInfo`:
I did this initially, but it created issues. Essentially now `onUiDisplayDataChanged` is for parsing new trajectories, whereas `selectionStateInfo` is about for responding to user changes. Unlikely to merge the paths although open to feedback and also maybe in need of renaming/documenting?